In [29]:
import pandas as pd
import matplotlib.pyplot as plt   
%matplotlib inline
import statsmodels.api as sm
from sklearn.model_selection import train_test_split
import numpy as np
import statsmodels.formula.api as smf

In [2]:
df = pd.read_csv('salary.csv')

In [3]:
df = df.dropna(how='any',axis=0)

In [4]:
train, test = train_test_split(df, test_size=0.3)

In [5]:
X_train = train['yearsworked']
y_train = train['salary']

In [6]:
model = sm.OLS(y_train,X_train).fit()

In [7]:
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:                 salary   R-squared:                       0.768
Model:                            OLS   Adj. R-squared:                  0.767
Method:                 Least Squares   F-statistic:                     1185.
Date:                Wed, 27 Mar 2019   Prob (F-statistic):          1.24e-115
Time:                        21:10:01   Log-Likelihood:                -4147.5
No. Observations:                 359   AIC:                             8297.
Df Residuals:                     358   BIC:                             8301.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
yearsworked  2840.8675     82.514     34.429      

### Does the model significantly predicts the dependent variable? Report the amount of variance explained (R^2) and significance value (p) to support your answer.

The model significantly predicts the dependent variable. This is shown from variance (R^2) 0.768(higher value generally indicates a better fit) with a p value of 0.000... (less than 0.05 is considered to be statistically significant)


### What percentage of the variance in employees’ salaries is accounted for by the number of years they have worked?
77.2%



### What does the unstandardized coefficient (B or 'coef' in statsmodels) tell you about the relationship between Years Worked and Salary?
The coeffecient is 2840.5341 this means that when salary increases by 1, the number of years worked increases by -3.787e-05

### What do the 95% confidence intervals [0.025, 0.975] mean?

### Calculate the expected salary for someone with 12 years’ work experience.

In [39]:
smresults = smf.ols('yearsworked ~ salary', X_train).fit()
X_train['pred'] = smresults.predict()

PatsyError: Error evaluating factor: NameError: name 'yearsworked' is not defined
    yearsworked ~ salary
    ^^^^^^^^^^^

### Calculate the expected salary for someone with 80 years’ work experience. Are there any problems with this prediction? If so, what are they?

### We have only looked at the number of years an employee has worked. What other employee characteristics might influence their salary?

### How does your model compare when running it on the test set - what is the difference in the Root Mean Square Error (RMSE) between the training and test sets?

In [9]:
X_test = test['yearsworked']
y_test= test['salary']

In [10]:
test_model = sm.OLS(y_test,X_test).fit()

In [11]:
print_test_model = test_model.summary()
print(print_test_model)

                            OLS Regression Results                            
Dep. Variable:                 salary   R-squared:                       0.738
Model:                            OLS   Adj. R-squared:                  0.736
Method:                 Least Squares   F-statistic:                     430.9
Date:                Wed, 27 Mar 2019   Prob (F-statistic):           2.38e-46
Time:                        21:10:01   Log-Likelihood:                -1789.7
No. Observations:                 154   AIC:                             3581.
Df Residuals:                     153   BIC:                             3585.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
yearsworked  2929.3532    141.113     20.759      